components/kyma-operator/pkg/kymaoperation/steps/install.go

```diff
@@ -0,0 +1,74 @@
+package steps
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/kyma-project/kyma/components/kyma-operator/pkg/kymasources"
+	"github.com/kyma-project/kyma/components/kyma-operator/pkg/overrides"
+)
+
+type installStep struct {
+	step
+	sourceGetter      kymasources.SourceGetter
+	overrideData      overrides.OverrideData
+	deleteWaitTimeSec uint32
+}
+
+// Run method for installStep triggers step installation via helm
+func (s installStep) Run() error {
+
+	chartDir, err := s.sourceGetter.SrcDirFor(s.component)
+	if err != nil {
+		return err
+	}
+
+	releaseOverrides, releaseOverridesErr := s.overrideData.ForRelease(s.component.GetReleaseName())
+
+	if releaseOverridesErr != nil {
+		return releaseOverridesErr
+	}
+
+	installResp, installErr := s.helmClient.InstallRelease(
+		chartDir,
+		s.component.Namespace,
+		s.component.GetReleaseName(),
+		releaseOverrides)
+
+	if installErr != nil {
+		installErrMsg := fmt.Sprintf("Helm install error: %s", installErr.Error())
+		errorMsg := installErrMsg
```

Comment:
Again, I don't think we need this duplication

Reply:
This was existing code that I did not care to refactor - these are problems local to this method and I was addressing more important (IMO) issues at package/module level. Having said that, I'll see what can be done here...